Assets/Scripts/Combat/CombatController.cs

```diff
@@ -10,57 +10,92 @@ public class CombatController : TileBlockerController
         ATTACK_ONLY,
         // Charges get 'free' movement points equal to Constants.ATTACK_AP_COST before the character starts paying for movement.
         // But, moves are only valid if they end in an attack.
         CHARGE_ATTACK,
-        // Ranged attacks wil raycast to the target, 
-        RANGED_ATTACK
+        // Ranged attacks wil raycast to the target, and this move MUST end in an attack.
+        RANGED_ATTACK_ONLY,
+        // Basic ranged attack or movement.
+        DEFAULT_RANGED
     };
 
+    public bool Dead()
+    {
+        return creatureMechanics.dead;
+    }
+
     private void RegisterMoves()
     {
         if (GetComponent<ActionRegenerate>() != null) specialMoves.Add(GetComponent<ActionRegenerate>());
         if (GetComponent<ActionBullRush>() != null) specialMoves.Add(GetComponent<ActionBullRush>());
         if (GetComponent<ActionSlaughter>() != null) specialMoves.Add(GetComponent<ActionSlaughter>());
         if (GetComponent<ActionRage>() != null) specialMoves.Add(GetComponent<ActionRage>());
+        if (GetComponent<ActionEmpower>() != null) specialMoves.Add(GetComponent<ActionEmpower>());
+        if (GetComponent<ActionOffhandAttack>() != null) specialMoves.Add(GetComponent<ActionOffhandAttack>());
+        if (GetComponent<ActionLifeOrDeath>() != null) specialMoves.Add(GetComponent<ActionLifeOrDeath>());
+        if (GetComponent<ActionMultiAttack>() != null) specialMoves.Add(GetComponent<ActionMultiAttack>());
+        if (GetComponent<ActionSnakeBite>() != null) specialMoves.Add(GetComponent<ActionSnakeBite>());
+        if (GetComponent<ActionPetrify>() != null) specialMoves.Add(GetComponent<ActionPetrify>());
+        if (GetComponent<ActionTailSweep>() != null) specialMoves.Add(GetComponent<ActionTailSweep>());
+        if (GetComponent<ActionTerrify>() != null) specialMoves.Add(GetComponent<ActionTerrify>());
     }
 
     override protected void Start()
     {
+        manager = transform.parent.GetComponent<TurnManager>();
         creatureMechanics = GetComponent<CreatureMechanics>();
         panel = Object.FindObjectOfType<GUIPanel>();
         PopupTextController.Initialize();
         RegisterMoves();
+        selectedAction = GetComponent<ActionBasicAttack>();
         base.Start();
     }
     
     public void AssignZonesOfControl()
     {
-        foreach (Tile adjacentTile in currentTile.adjacentTileList)
+        foreach (Tile adjacentTile in AdjacentTiles())
         {
             adjacentTile.SetIsZoneOfControl(true);
         }
     }
 
+    public List<Tile> AdjacentTiles()
+    {
+        return currentTile.adjacentTileList;
+    }
+
+    private void CooldownSpecialMoves()
+    {
+        foreach (Action a in specialMoves)
+        {
+            a.AdvanceCooldown();
+        }
+    }
+
     public void BeginTurn()
     {
-        
+        CooldownSpecialMoves();
         actionPoints = creatureMechanics.BeginTurnAndGetMaxActionPoints();
+        if (Dead()) return;
```

Comment:
I would move this at the top of the method.
If (dead) nothing else needs to be computed (CooldownSpecialMoves or actionPoints)

Reply:
The check has to be after begin turn, because mechanic's begin turn can kill the unit (for example, poison effects at the beginning of turn).